implementation/src/main/java/io/smallrye/metrics/exporters/OpenMetricsExporter.java

```diff
@@ -364,15 +376,19 @@ private void addTags(StringBuilder sb, Map<String, String> tags) {
     }
 
     private void fillBaseName(StringBuilder sb, MetricRegistry.Type scope, String key, String suffix) {
-        sb.append(scope.getName().toLowerCase()).append("_").append(key);
+        if (usePrefixForScope) {
+            sb.append(scope.getName().toLowerCase()).append("_");
+        }
+        sb.append(key);
         if (suffix != null)
             sb.append(suffix);
     }
 
     private void writeHelpLine(final StringBuilder sb, MetricRegistry.Type scope, String key, Metadata md, String suffix) {
         // Only write this line if we actually have a description in metadata
         Optional<String> description = md.getDescription();
-        if (writeHelpLine && description.filter(s -> !s.isEmpty()).isPresent() && !alreadyExportedNames.get().contains(md.getName())) {
```

Comment:
I faced that issue in WildFly while integrating the latest smallrye-metrics. The `wildfly_undertow_request_count_total` metric can appear in 3 different places (http-listener, https-listener, servlet) so I had to check whether the metric name is already registered or not. I have to use the same description for the 3 metrics instead of being more specific but that's not an actual issue as the metric names should be already quite self-descriptive.

Note that I named the configuration setting `"smallrye.metrics.usePrefixForScope"` and not `"microprofile.metrics.usePrefixForScope"` to make it clear that this is a smallrye-metrics feature, not a MicroProfile one (i.e. we accept that there will be some limitations regarding metric names/descriptions that are worked around by having sensible metric names)

Reply:
Ok let's leave it as it is for now. I'll create a note to document this behavior once we have any actual documentation in place :)